rpc/src/p2p.rs

```diff
@@ -1,17 +1,207 @@
 use jsonrpsee::proc_macros::rpc;
-use serde::{Deserialize, Serialize};
+use libp2p::Multiaddr;
+use serde::{de, de::Error, ser, Deserialize, Serialize};
+use std::collections::HashMap;
+use std::str::FromStr;
+use thiserror::Error;
 
 #[derive(Debug, Serialize, Deserialize)]
 pub struct AddrInfo {
     #[serde(rename = "ID")]
-    pub id: String,
-    // TODO: multiaddr
+    pub id: PeerId,
     #[serde(rename = "Addrs")]
-    pub addrs: Vec<String>,
+    pub addrs: Vec<Multiaddr>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Stat {
+    #[serde(rename = "NumStreamsInbound")]
+    pub streams_inbound: u32,
+    #[serde(rename = "NumStreamsOutbound")]
+    pub streams_outbound: u32,
+    #[serde(rename = "NumConnsInbound")]
+    pub connections_inbound: u32,
+    #[serde(rename = "NumConnsOutbound")]
+    pub connections_outbound: u32,
+    #[serde(rename = "NumFD")]
+    pub file_descriptors: u32,
+    #[serde(rename = "Memory")]
+    pub memory: u32,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ResourceManagerStats {
+    #[serde(rename = "System")]
+    pub system: Stat,
+    #[serde(rename = "Transient")]
+    pub transient: Stat,
+    #[serde(rename = "Services")]
+    pub services: HashMap<String, Stat>,
+    #[serde(rename = "Protocols")]
+    pub protocols: HashMap<String, Stat>,
+    #[serde(rename = "Peers")]
+    pub peers: HashMap<String, Stat>,
+}
```

Comment:
use [`rename_all`](https://serde.rs/container-attrs.html#rename_all) for the field names

Reply:
done, note that these struct have been moved to `types`.